experimental/dds/tree2/docs/Shared Tree Philosophy.md

```diff
@@ -0,0 +1,194 @@
+# Shared Tree
+
+The goal of the shared tree project is empower developers to create and maintain better collaborative experiences.
+
+To best achieve this goal we plan to deliver a set of libraries that enable developers unfamiliar with the challenges specific to collaborative applications to:
+
+-   Adopt Shared Tree without concern for having to migrate to another solution due to hitting limitations.
+
+    This means that if a developer wants to make a change that interacts with collaboration, the fact that their application is powered by shared tree will not be a limiting factor on what can be achieved.
+    This includes compatibility between different application or version of an applications, scalability, availability, merge quality etc.
+    More concretely, if faced with a new application collaboration requirement, it should be practical to either build it on top of shared tree or extend shared tree to support it.
+    Doing this should not require major reworking of either the application or shared tree implementations, not require extending it in a way that the upstream shared tree would not be willing to maintain.
+
+-   Productively create and maintain collaborative experiences.
+
+    Doing this "productively" means ensuring that a developer new to collaboration will be directed into using patterns that work well for collaborative applications,
+    and ensuring that useful supporting libraries are in place to make this a good experience.
+
+-   Avoid common pitfalls of collaborative software.
+
+    Collaboration often introduces subtle requirements, especially around compatibility, updates, offline use, error cases, persistance and concurrency.
+    Picking the right design patterns (above) as well has explicitly drawing attention to particular details in the API design, documentation and examples all need to work together to holistically guide developers into creating experiences that collaborate correctly.
+    Additionally, the resulting experiences need to be maintainable, including easy authoring of new features and reviewing of changes for possible collaboration issues.
+    For code authors, it should be easy to find the right way to do something, and know it will be robust.
+    For code reviews, it should be obvious in review, only looking at the part of the code change, if it might introduce a collaboration related issue (such as unexpected merge behavior, issues collaborating with previous or future versions of the app, inability to open old document etc.).
```

Comment:
```suggestion
    For code reviews, it should be obvious in review, only looking at the changed code, if it might introduce a collaboration related issue (such as unexpected merge behavior, issues collaborating with previous or future versions of the app, inability to open old document, etc.).
```